doc/plugin-criteria.md

```diff
@@ -0,0 +1,129 @@
+           </ul>
+        </td>
+    </tr>
+    <tr>
+        <td>2.</td>
+        <td>
+          The task provides a different solution to a problem which is also solved by kubectl.
+          However, the task would be impossible or very hard to reproduce with kubectl alone.
+        </td>
+        <td>
+          In general welcome unless it overlaps significantly with other plugins or has a bad usability.
+          Examples:
+          <ul>
+            <li><code>get-all</code> plugin is a replacement for <code>kubectl get all</code> command which actually fetches only a subset of API resource types.</li>
+            <li><code>access-matrix</code> gives an overview of authorities, whereas <code>kubectl auth can-i</code> only applies to single resource/verb pairs</li>
+            <li><code>tail</code> plugin allows tailing logs from pods from a label selector, a namespace or all namespaces at once, whereas <code>kubectl logs</code> command has more limited functionality.</li>
+            <li><code>mtail</code> plugin allows tailing logs from pods from a label selector, similar to <code>kubectl logs -l</code> but labels log lines.</li>
+          </ul>
+        </td>
+    </tr>
+    <tr>
+        <td>3.</td>
+        <td>The task addresses a completely different problem than any of the kubectl subcommands, but is clearly focused on k8s.</td>
+        <td>Gray area. If the task fits well into <code>kubectl</code>, it may be accepted (for example <code>kubectl debug-pod</code>).</td>
+    </tr>
+    <tr>
+        <td>4.</td>
+        <td>
+          The task is very dissimilar to common kubectl tasks and is not primarily focused on k8s.
+          However, it can be useful in the k8s ecosystem.
+        </td>
+        <td>Not acceptable.</td>
+    </tr>
+    <tr>
+        <td>5.</td>
+        <td>
+          The task is very dissimilar to common kubectl tasks and is not focused on k8s.
+          Its general usefulness in the k8s ecosystem is questionable.
+        </td>
+        <td>Not acceptable.</td>
+    </tr>
+    </tbody>
+</table>
+
+Category
+---
+
+In order of decreasing usage frequency:
+
+<table>
+    <thead>
+    <tr>
+        <th>#</th>
+        <th>Classification</th>
+        <th>Recommendation</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr>
+        <td>1.</td>
+        <td>Recurring task on the same cluster. For example: <code>mtail</code></td>
+        <td>Favors acceptance as plugin.</td>
+    </tr>
+    <tr>
+        <td>2.</td>
+        <td>
+          Recurring task on different cluster instances, but one-time task on the same cluster instance.
+          For example: <code>init-tiller</code>
+        </td>
+        <td>Gray area. Usually this is not acceptable.</td>
+    </tr>
+    <tr>
+        <td>3.</td>
+        <td>One-time task per machine.</td>
+        <td>In general not acceptable as plugin.</td>
+    </tr>
+    </tbody>
+</table>
+
+Usability
+---
+Further required acceptance criteria:
```

Comment:
I think we should relax this part a bit, and maybe make it optional to not to limit creativity too much.

Reply:
Do you meant to not talk about "required" criteria?

Reply:
I meant we should just remove the language that makes it "required". We can still list best practices, or develop a separate style guide.

Reply:
ok, makes sense.